portality/lib/plausible.py

```diff
@@ -40,8 +41,12 @@ def send_event(goal: str, on_completed=None, **props_kwargs):
     if props_kwargs:
         payload['props'] = json.dumps(props_kwargs)
 
+    headers = {}
+    if request:
+        headers = {"X-Forwarded-For": request.remote_addr}  # this works because we have ProxyFix on the app
```

Comment:
Our current ProxyFix setup is `ProxyFix(app.wsgi_app, x_proto=1, x_host=1)`, which corresponds to our nginx directives:

```
proxy_set_header Host $host;
proxy_set_header X-Real-IP $remote_addr;   [adds cloudflare's IP]
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;    [adds Cloudflare's IP]
proxy_set_header X-Forwarded-Proto $scheme;
```

We have to be careful here to make these custom events equivalent to the frontend - will `request.remote_addr` be the real IP if we change the nginx request to include `$http_cf_connecting_ip` - particularly if we're appending rather than replacing as per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#syntax

So what I'm asking here is:
* does ProxyFix / flask make any assumptions to calculate request.remote_addr that will differ from the main Plausible client?
* do we need to do anyting with X-Real-IP in addition to forwarded_for
* can cloudflare announce its existence to complate the chain in x-forwarded_for or do we need to do that ourselves in nginx?

Reply:
I don't know the answers to any of that, I think we'll basically need to test it for real and see what happens.  Perhaps we could set up a mock Plausible endpoint for testing, and log what gets sent.

From the documentation, I got the impression that that the point of ProxyFix was to set `request.remote_addr` to the correct value whatever the situation, although presumably as you say the configs need to match.